File: Core/src/main/java/dev/lynith/core/utils/nvg/Font.java
package dev.lynith.core.utils.nvg;

import lombok.Getter;
import lombok.Setter;

import java.nio.ByteBuffer;

@Setter @Getter
public class Font {

    public final String name;
    public final String path;
    public final FontWeight weight;
    public ByteBuffer buffer;
    public final float offset;

    public Font(String name, String path, FontWeight weight, float offset) {
        this.name = name;
        this.path = path;
        this.weight = weight;
        this.offset = offset;
    }

    public Font(String name, String path, FontWeight weight) {
        this(name, path, weight, 0);
    }

    public String formatted() {
        return name + "-" + weight.formatted();
    }

    @Override
    public String toString() {
        return "Font{" +
            "name='" + name + '\'' +
            ", path='" + path + '\'' +
            ", weight=" + weight +
            ", offset=" + offset +
            '}';
    }

    public enum FontWeight {
        LIGHT("Light"),
        REGULAR("Regular"),
        MEDIUM("Medium"),
        BOLD("Bold");

        private final String name;

        public String formatted() {
            return name;
        }

        FontWeight(String name) {
            this.name = name;
        }
    }

    public enum FontAlign {
        LEFT,
        CENTER,
        RIGHT,
    }

}

File: Core/src/main/kotlin/dev/lynith/core/ui/nvg/Font.kt
package dev.lynith.core.ui.nvg

import java.nio.ByteBuffer

class Font @JvmOverloads constructor(
    val name: String,
    val path: String,
    val weight: FontWeight,
    val offset: Float = 0f
) {
    var buffer: ByteBuffer? = null

    fun formatted(): String {
        return "$name-$weight"
    }

    override fun toString(): String {
        return "Font{" +
            "name='" + name + '\'' +
            ", path='" + path + '\'' +
            ", weight=" + weight +
            ", offset=" + offset +
            '}'
    }

    enum class FontWeight(private val weight: String) {
        LIGHT("Light"),
        REGULAR("Regular"),
        MEDIUM("Medium"),
        SEMIBOLD("SemiBold"),
        BOLD("Bold"),
        BLACK("Black");

        override fun toString(): String {
            return weight
        }
    }

    enum class FontAlign {
        LEFT,
        CENTER,
        RIGHT
    }
}
